src/Symfony/Bundle/WebServerBundle/DependencyInjection/WebServerExtension.php

```diff
@@ -54,4 +60,16 @@ private function getPublicDirectory(ContainerBuilder $container)
 
         return $kernelProjectDir.'/'.$publicDir;
     }
+
+    private function getPidFileDirectory(ContainerBuilder $container)
+    {
+        $kernelCacheDir = $container->getParameter('kernel.cache_dir');
+        $environment = $container->getParameter('kernel.environment');
+
+        if (basename($kernelCacheDir) !== $environment) {
```

Comment:
I think we should use the `kernel.project_dir` only in case someone is not following the cache folder structure convention.